File: cpp/Student.cpp
// import liblary
#include <iostream>
#include <string>
#include <iomanip>

// deklarasi std agar tidak ditulis ulang
using namespace std;

// kelas mahasiswa, untuk menyimpan data mahasiswa
class Student{
    private:
        // memiliki atribut sebagai berikut, dengan semua tipe data string
        string NIM;
        string name;
        string prody;
        string faculty;

    public:
        // construktor
        Student(string name, string NIM, string prody, string faculty){
            // memiliki parameter untuk set atribut dalam kelas 
            this->name = name;
            this->NIM = NIM;
            this->prody = prody;
            this->faculty = faculty;
        }

        // method untuk menampilkan informasi mahasiswa
        void displayStudentInfo(){
            // setting untuk panjang kolom
            int tabLen[4] = {25, 15, 25, 15};

            // output untuk tampilan
            cout << left << setw(tabLen[0]) << "| " + name << setw(tabLen[1]) << "| " + NIM << 
            setw(tabLen[2]) << "| " + prody << setw(tabLen[3]) << "| " + faculty << "|" << endl;
        }

        // dibawah merupakan kumpulan method setter dan getter dari setiap atribut
        void setName(string name){
            this->name = name;
        }

        string getName(){
            return name;
        }

        void setNIM(string NIM){
            this->NIM = NIM;
        }

        string getNIM(){
            return NIM;
        }

        void setPrody(string prody){
            this->prody = prody;
        }

        string getPrody(){
            return prody;
        }

        void setFaculty(string faculty){
            this->faculty = faculty;
        }

        string getFaculty(){
            return faculty;
        }

        // destruktor
        ~Student(){

        }
};

File: java/Student.java
// kelas mahasiswa, untuk menyimpan data mahasiswa
class Student {
    // memiliki atribut sebagai berikut, dengan semua tipe data string
    private String NIM;
    private String name;
    private String prody;
    private String faculty;

    // construktor
    public Student(String name, String NIM, String prody, String faculty) {
        // memiliki parameter untuk set atribut dalam kelas 
        this.name = name;
        this.NIM = NIM;
        this.prody = prody;
        this.faculty = faculty;
    }

    // method untuk menampilkan informasi mahasiswa
    public void displayStudentInfo(int[] tabLen) {
        // output untuk tampilan
        System.out.format("| %-" + (tabLen[0]-2) + "s| %-" + (tabLen[1]-2) + "s| %-" + (tabLen[2]-2) + "s| %-" + (tabLen[3]-2) + "s|\n", name, NIM, prody, faculty);
    }

    // dibawah merupakan kumpulan method setter dan getter dari setiap atribut
    public void setName(String name) {
        this.name = name;
    }

    public String getName() {
        return name;
    }

    public void setNIM(String NIM) {
        this.NIM = NIM;
    }

    public String getNIM() {
        return NIM;
    }

    public void setPrody(String prody) {
        this.prody = prody;
    }

    public String getPrody() {
        return prody;
    }

    public void setFaculty(String faculty) {
        this.faculty = faculty;
    }

    public String getFaculty() {
        return faculty;
    }
}